### PR TITLE
Refactor PLE and PLP

### PR DESCRIPTION
**Pull request by anthonysena**
Per #988, this PR aims to address feedback from #953 and some items from #988. This code should be tested with OHDSI/WebAPI#738 and OHDSI/StandardizedAnalysisAPI#14. This PR is also quite large so I'm happy to review this via a call if easier.

Some notes for discussion:

- I've done my best to follow the BEM methodology and have used LESS. Would appreciate any feedback from @johnSamilin and @pavgra on this.
- Per this [comment](https://github.com/OHDSI/Atlas/pull/953#discussion_r221873072), I cannot move this method to the class level since the binding to the data table fails to work once I do that.
- I'm not clear on why I should change the code for this [comment](https://github.com/OHDSI/Atlas/pull/953#discussion_r221873130). 
- I'm unsure why I should [split this into 2 separate methods](https://github.com/OHDSI/Atlas/pull/953#discussion_r221873570) - I need to check if either the button or the inner target are clicked to trigger the delete event.
- I can't use [lodash.range(10,0)](https://github.com/OHDSI/Atlas/pull/953#discussion_r223035470) since the editable drop down requires strings as input. I made an attempt to use lodash but found converting to strings to be a bit hacky so I kept the code as is.
- For the comments: [here](https://github.com/OHDSI/Atlas/pull/953#discussion_r223036534) and [here](https://github.com/OHDSI/Atlas/pull/953#discussion_r223036927), I've left the code the same since I find it easier to read and to compare against the method signatures in the FeatureExtraction methods library.

There is still more to do to fully address #988 but given the large number of changes so far, I thought it best to get this into through the review process and then address the remaining items after that.

Thanks!

### CODE REVIEW

File: js/pages/estimation/const.js [hunk 1]
```diff
@@ -37,10 +48,363 @@ define(
...
     };
 
+    const getTimeAtRisk = (createStudyPopArgs) => {
+        return (createStudyPopArgs.riskWindowStart() + "-" + createStudyPopArgs.riskWindowEnd() + "d<br/>(min: " + createStudyPopArgs.minDaysAtRisk() + "d)");
+    };
+
+    const options = {
+      removeButton: `<button type="button" class="btn btn-danger btn-xs btn-remove"><i class="fa fa-times" aria-hidden="true"></i></button>`,
+      copyButton: `<button type="button" class="btn btn-primary btn-xs btn-copy"><i class="fa fa-clone" aria-hidden="true"></i>&nbsp;Copy</button>`,
+      numberOfStrataOptions: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'],
```
[the remark this answers is not in the document]
Comment (anthonysena) [reply js/pages/estimation/const.js]: Sorry, my comment in this PR was not accurate here. This array is used to populate the editable drop down list component. It is expecting an array of single quote strings which I can't seem to get to work with lodash:

```
_.range(1, 11).map(v => '' + v)
(10) ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"]
```

[the remark this answers is not in the document]
Comment (pavgra) [reply js/pages/estimation/const.js]: >  single quote strings

What's the difference in how a string is printed out or defined? (We are not in PHP)

[the remark this answers is not in the document]
Comment (pavgra) [reply js/pages/estimation/const.js]: >  single quote strings

What's the difference in how a string is printed out or defined? (We are not in PHP)

[the remark this answers is not in the document]
Comment (anthonysena) [reply js/pages/estimation/const.js]: I am unsure why it has an impact but it does. We can review it together on Friday to make sure I'm not mistaken here.

**Comment by anthonysena** [comment]
@pavgra from our review and comments here:

- Shared data classes were moved from `js/components/analysis` to `js/services/analysis`
- lodash is being used to populate arrays of numbers versus verbatim lists
- Styles in .less are using kebab case
- Updated components with data table binding functions in the constructor to extend `AutoBind(Component)`  instead of `Component` which enabled me to move those functions out of the constructor.
- Utilizing async/await for deletion methods for PLE/PLP

In my view, this PR is ready but should not be merged into master until OHDSI/WebAPI#738 and OHDSI/StandardizedAnalysisAPI#14 are approved. I will be making updates to those PRs based on our code review shortly. Thanks!

**Comment by pavgra** [comment]
@anthonysena , I would drop legacy PLE / PLP files since those are going to confuse people during development - two sets of files next to each other in the same folders. Also, could you please merge in latest master?

**Comment by anthonysena** [comment]
@pavgra - while merging in the latest master, I've noticed that the dirtyFlag for estimation & prediction is not functioning properly. I'll need some time tomorrow to investigate but suspect it is related to #1321. WebAPI and StdAPI are merged in - just need to clean out the old files as discussed.

**Comment by pavgra** [comment]
@anthonysena , the dirty flag issue (https://github.com/OHDSI/Atlas/issues/1321) was resolved - the fix is already available in the latest master

**Pull request merged**